### PR TITLE
Tests: Fix Circle CI randomly failing on one of the containers when executing client tests

### LINKED ISSUE #4844: Tests: Circle CI randomly failing on one of the containers when executing client tests
During last 3 days we have seen at least 5 failed build on one of the containers:
* https://circleci.com/gh/Automattic/wp-calypso/13480
* https://circleci.com/gh/Automattic/wp-calypso/13585
* https://circleci.com/gh/Automattic/wp-calypso/13603
* https://circleci.com/gh/Automattic/wp-calypso/13615
* https://circleci.com/gh/Automattic/wp-calypso/13662

There are 4 unique errors that were noticed:
screen shot 2016-04-19 at 07 03 16

The command that caused container to fail looks like:
`npm run test-client 'client/lib/happiness-engineers/test/actions.js' 'client/state/users/test/selectors.js' 'client/lib/mixins/searchable/test/index.js' 'client/state/analytics/test/actions.js' 'client/state/google-apps-users/test/actions.js' 'client/state/post-types/test/reducer.js' 'client/components/accordion/test/index.jsx' 'client/lib/reader-lists-items/test/store.js' 'client/state/ui/editor/contact-form/test/actions.js' 'client/lib/plugins/wporg-data/test/test-actions.js' 'client/devdocs/test/doc.jsx' 'client/lib/media-serialization/test/index.js' 'client/components/drop-zone/test/index.jsx' 'client/lib/media/test/validation-store.js' 'client/lib/site/test/index.js' 'client/lib/security-checkup/test/actions.js' 'client/state/support/test/actions.js' 'client/lib/keyboard-shortcuts/test/key-bindings.js' 'client/lib/invites/stores/test/invites-list.js' 'client/lib/happiness-engineers/test/store.js' 'client/lib/security-checkup/test/constants.js' 'client/lib/purchases/stored-cards/test/store.js' 'client/state/comments/test/actions.js' 'client/components/feature-example/test/test.js' 'client/state/themes/themes/test/reducer.js' 'client/state/plugins/wporg/test/actions.js' 'client/lib/analytics/page-view-tracker/test/index.jsx' 'client/lib/signup/test/dependency-store.js' 'client/state/themes/current-theme/test/reducer.js' 'client/state/comments/test/selectors.js' 'client/my-sites/menus/item-options/test/posts.jsx' 'client/signup/config/test/index.js' 'client/lib/sites-list/test/log-store.js' 'client/state/themes/current-theme/test/index.js' 'client/lib/terms/test/category-store.js' 'client/lib/post-formats/test/store.js' 'client/lib/invites/reducers/test/invites-create-validation.js' 'client/my-sites/media-library/test/list.jsx' 'client/lib/popup-monitor/test/index.js' 'client/state/comments/test/reducer.js' 'client/lib/reader-site-store/test/index.js' 'client/lib/users/test/store.js' 'client/lib/email-followers/test/store.js' 'client/lib/stats/stats-list/test/stats-parser.js' 'client/lib/sites-list/test/index.js' 'client/lib/stats/stats-list/test/index.js' 'client/lib/site/test/utils.js' 'client/state/site-settings/exporter/test/actions.js' 'client/state/post-types/test/selectors.js' 'client/state/ui/editor/last-draft/test/selectors.js' 'client/state/posts/counts/test/actions.js' 'client/components/tinymce/plugins/contact-form/test/shortcode.js' 'client/components/forms/counted-textarea/test/index.jsx' 'client/lib/purchases/test/store.js' 'client/lib/interval/test/component.jsx' 'client/signup/test/flows.js' 'client/lib/phone-validation/test/index.jsx' 'client/components/tinymce/plugins/contact-form/test/validations.js' 'client/state/plugins/wporg/test/reducer.js' 'client/state/ui/editor/last-draft/test/reducer.js' 'client/lib/tree-convert/test/index.js' 'client/lib/recommended-sites-store/test/actions.js' 'client/post-editor/media-modal/test/preload-image.js' 'client/post-editor/media-modal/test/index.jsx' 'client/state/ui/reader/fullpost/test/reducer.js' 'client/state/comments/test/utils.js' 'client/lib/mixins/pageable/test/index.js' 'client/post-editor/test/post-editor.jsx' -- -w`

### PR DESCRIPTION
**Pull request by gziolo**
Fixes #4844.

It makes sure we don't see 4 unique errors anymore on Circle CI:
screen shot 2016-04-19 at 07 03 16

There were 2 root causes:
* `<InfiniteList />` executing async callbacks that were interfering with the next test file
* there were some issues with `i18n` initialisation

### Testing
Run server times 2 following commands and make sure there are no test failures:

##### All tests
`npm run test-client`

#### Tests executed on the problematic container
`npm run test-client 'client/lib/happiness-engineers/test/actions.js' 'client/state/users/test/selectors.js' 'client/lib/mixins/searchable/test/index.js' 'client/state/analytics/test/actions.js' 'client/state/google-apps-users/test/actions.js' 'client/state/post-types/test/reducer.js' 'client/components/accordion/test/index.jsx' 'client/lib/reader-lists-items/test/store.js' 'client/state/ui/editor/contact-form/test/actions.js' 'client/lib/plugins/wporg-data/test/test-actions.js' 'client/devdocs/test/doc.jsx' 'client/lib/media-serialization/test/index.js' 'client/components/drop-zone/test/index.jsx' 'client/lib/media/test/validation-store.js' 'client/lib/site/test/index.js' 'client/lib/security-checkup/test/actions.js' 'client/state/support/test/actions.js' 'client/lib/keyboard-shortcuts/test/key-bindings.js' 'client/lib/invites/stores/test/invites-list.js' 'client/lib/happiness-engineers/test/store.js' 'client/lib/security-checkup/test/constants.js' 'client/lib/purchases/stored-cards/test/store.js' 'client/state/comments/test/actions.js' 'client/components/feature-example/test/test.js' 'client/state/themes/themes/test/reducer.js' 'client/state/plugins/wporg/test/actions.js' 'client/lib/analytics/page-view-tracker/test/index.jsx' 'client/lib/signup/test/dependency-store.js' 'client/state/themes/current-theme/test/reducer.js' 'client/state/comments/test/selectors.js' 'client/my-sites/menus/item-options/test/posts.jsx' 'client/signup/config/test/index.js' 'client/lib/sites-list/test/log-store.js' 'client/state/themes/current-theme/test/index.js' 'client/lib/terms/test/category-store.js' 'client/lib/post-formats/test/store.js' 'client/lib/invites/reducers/test/invites-create-validation.js' 'client/my-sites/media-library/test/list.jsx' 'client/lib/popup-monitor/test/index.js' 'client/state/comments/test/reducer.js' 'client/lib/reader-site-store/test/index.js' 'client/lib/users/test/store.js' 'client/lib/email-followers/test/store.js' 'client/lib/stats/stats-list/test/stats-parser.js' 'client/lib/sites-list/test/index.js' 'client/lib/stats/stats-list/test/index.js' 'client/lib/site/test/utils.js' 'client/state/site-settings/exporter/test/actions.js' 'client/state/post-types/test/selectors.js' 'client/state/ui/editor/last-draft/test/selectors.js' 'client/state/posts/counts/test/actions.js' 'client/components/tinymce/plugins/contact-form/test/shortcode.js' 'client/components/forms/counted-textarea/test/index.jsx' 'client/lib/purchases/test/store.js' 'client/lib/interval/test/component.jsx' 'client/signup/test/flows.js' 'client/lib/phone-validation/test/index.jsx' 'client/components/tinymce/plugins/contact-form/test/validations.js' 'client/state/plugins/wporg/test/reducer.js' 'client/state/ui/editor/last-draft/test/reducer.js' 'client/lib/tree-convert/test/index.js' 'client/lib/recommended-sites-store/test/actions.js' 'client/post-editor/media-modal/test/preload-image.js' 'client/post-editor/media-modal/test/index.jsx' 'client/state/ui/reader/fullpost/test/reducer.js' 'client/state/comments/test/utils.js' 'client/lib/mixins/pageable/test/index.js' 'client/post-editor/test/post-editor.jsx' -- -w`

### CODE REVIEW

File: client/my-sites/media-library/test/list.jsx [hunk 1]
```diff
@@ -55,7 +51,8 @@ describe( 'MediaLibraryList item selection', function() {
 			data: fixtures
 		} );
 
-		mockery.registerMock( './list-item', EMPTY_COMPONENT );
+		mockery.registerMock( 'components/infinite-list', EmptyComponent );
```
Comment (gziolo) [inline client/my-sites/media-library/test/list.jsx]: `<InfiniteList />` have async callbacks which cause issues. We don't verify it's logic here so I mocked it out :)

**Comment by blowery** [comment]
Looks good to me, but @aduth should have the final call.

**Comment by aduth** [comment]
Encountered a failure on `master` branch with the command in the original comment on my second attempt. Could not hit failure after a dozen attempts on `fix/4844-circle-ci-failures`.

Media changes look fine. Kinda a shame we go through the whole mounting procedure just to test a few utility methods on the instance.

Were the changes to `useI18n` necessary to fixing the failures, or just cleanup?

I'm not sure what specifically it is about `<InfiniteList />` which would cause test failures, but we may want to circle back around to it in the future, as not all tests may be so easily able to mock it with an empty component.

Good to merge, I'd say 👍

**Comment by gziolo** [comment]
> Were the changes to useI18n necessary to fixing the failures, or just cleanup?

To be honest. I'm not quite sure :) I started fixing failures going from the top of the list and that was working solution. It might be not necessary to useI18n, but once I refactored it, I thought we better keep it as it's cleaner solution.

**Comment by gziolo** [comment]
> I'm not sure what specifically it is about <InfiniteList /> which would cause test failures

I'm not familiar with this components, but there is something tricky with the way `didComponentMount` works. It does lots of magic, sets events, checks scroll position. Quite advanced logic.

**Comment by aduth** [comment]
>I'm not familiar with this components [...] It does lots of magic

I think this sums our collective feeling toward the `<InfiniteList />` component 😆

**Comment by gwwar** [comment]
InfiniteList is extremely complex and not that performant. :) Thanks for looking at this @gziolo

**Pull request merged**